partcad/src/partcad/test/test.py

```
@@ -21,30 +22,33 @@ class Test(ABC):
 
     def __init__(self, name: str) -> None:
         self.name = name
-
+        self.cache_tests_lock = asyncio.Lock()
+        
     async def test_cached(self, tests_to_run: list["Test"], ctx, shape, test_ctx: dict = {}) -> bool:
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)